pennylane/ops/qubit.py

```diff
@@ -1502,6 +1583,10 @@ def decomposition(phi, lam, wires):
         ]
         return decomp_ops
 
+    def adjoint(self, do_queue=False):
+        # TODO(chase): Replace the `inv()` by instead modifying the parameters.
+        return U2(*self.parameters, wires=self.wires, do_queue=do_queue).inv()
```

Comment:
The inverse here is?
```
R(\lambda,-\pi/2,-\lambda) R_\phi(-\phi-\lambda)
```

Could we just add that now?

Reply:
What's R and R_phi in this context?

Reply:
Oops, I wonder why those symbols are used in the docstring. Here, `R` is `Rot`, and `R_phi` is `PhaseShift`.

Reply:
Adjoint right now can only return a single gate. Can we not just modify the `self.parameters` so that the new U2 gate is the adjoint of the original? (What actually is the adjoint math for the U{1,2,3} gates?)

Reply:
> Can we not just modify the self.parameters so that the new U2 gate is the adjoint of the original?

Probably, let me try and work it out.

> Adjoint right now can only return a single gate.

This feels like an odd restriction to me 🤔 What if the adjoint of a gate cannot be represented by a single gate?

Reply:
What single gate doesn't also have a single gate adjoint?

For some transforms I agree with you that they should return multiple gates, but I would prefer we keep adjoint simple.